src/app/core/components/event-card/event-card.component.spec.ts

```diff
@@ -0,0 +1,50 @@
+import { DebugElement } from '@angular/core';
+import { MatIconModule } from '@angular/material/icon';
+import { MatCardModule } from '@angular/material/card';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { Event } from '../../../../backend-client/model/event';
+
+import { EventCardComponent } from './event-card.component';
+
+describe('EventCardComponent', () => {
+  let component: EventCardComponent;
+  let fixture: ComponentFixture<EventCardComponent>;
+
+  const MOCK_EVENT_OBJECT: Event = {
+    name: 'Mock Test Event 1',
+    description: 'This is a event abot mocking',
+    organizationID: 'Chula.Org',
+    tags: ['Hello', 'World'],
+    startDate: ('2020-06-13T13:54:29.808Z' as unknown) as Date,
+    endDate: ('2020-06-13T13:54:29.808Z' as unknown) as Date,
+  };
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [EventCardComponent],
+      imports: [MatCardModule, MatIconModule],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(EventCardComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should display Event Name, Start Date and End Date correctly', () => {
```

Comment:
Optional: Having unit test is GOOD, but having unit test with 100% [code coverage ](https://medium.com/@krishankantsinghal/how-to-read-test-coverage-report-generated-using-jest-c2d1cb70da8b)is PHENOMENAL!. Try running yarn test with `--code-coverage` tag to check whether your unit test code covered all cases.

warning: code coverage 100% doesn't mean your code is 100% tested. it just mean jest think your code is 100% tested, and it can sometimes be pretty inaccurate. so it is advised to rely on your intuition about test coverage more than `--code-coverage` tag. TLDR: don't overused code coverage analysis

Reply:
I already read the links in your comment and search some more... I still don't get it well I think... what is that I should have done anyway...?